### PR TITLE
Replace functionality in deprecated lifecycle methods

### PR DESCRIPTION
**Pull request by ChrisDobby**
Removed componentWillReceiveProps from the following files:
components/searchable_channel_list.jsx
components/new_channel_flow/new_channel_flow.jsx
components/new_channel_modal/new_channel_modal.jsx
components/edit_channel_header_modal/edit_channel_header_modal.jsx
components/change_url_modal/change_url_modal.jsx
components/more_direct_channels/more_direct_channels.jsx

Fixes https://github.com/mattermost/mattermost-server/issues/12342

### CODE REVIEW

**Comment by jasonblais** [comment]
@hmhealey @bradjcoughlin Quick nudge to help with reviews on this PR :)

File: components/edit_channel_header_modal/edit_channel_header_modal.jsx [hunk 1]
```diff
@@ -68,6 +68,15 @@ class EditChannelHeaderModal extends React.PureComponent {
         }).isRequired,
     }
 
+    static getDerivedStateFromProps(props) {
+        const {requestStatus} = props;
+        if (requestStatus === RequestStatus.SUCCESS) {
+            return {show: false, showError: false};
+        }
+
+        return {showError: requestStatus === RequestStatus.FAILURE};
```
Comment (hmhealey) [inline components/edit_channel_header_modal/edit_channel_header_modal.jsx]: Could you remove `this.state.showError`? In the render method, we can just control the error text by whether or not `this.props.serverError` is set

File: components/searchable_channel_list.jsx [hunk 2]
```diff
@@ -19,6 +19,14 @@ import {t} from 'utils/i18n';
 const NEXT_BUTTON_TIMEOUT_MILLISECONDS = 500;
 
 export default class SearchableChannelList extends React.Component {
+    static getDerivedStateFromProps(props) {
+        if (!props.isSearch) {
```
Comment (hmhealey) [inline components/searchable_channel_list.jsx]: This needs to compare the previous value of `props.isSearch` against the previous one or else random prop updates might send the user back to the first page of results

**Comment by ChrisDobby** [comment]
Have updated the PR to reflect the review comments

**Comment by hanzei** [comment]
Thanks for the contribution @ChrisDobby :tada:

**Pull request merged**